src/templates_hdf5/templator_hdf5.org

```diff
@@ -497,7 +497,8 @@ trexio_hdf5_write_$group_dset$ (trexio_t* const file,
   trexio_hdf5_t* f = (trexio_hdf5_t*) file;
 
   hid_t index_dtype;
-  void* index_p = NULL;
+  const void* index_p = NULL;
```

Comment:
Could you clarify a bit why we need 2 void pointers here instead of one as it was implemented before?

Reply:
`index_sparse` is a `const int32_t *`. 
What we did before was to create a new array for smaller types, and use the same pointer if we are in 32 bits.
The problem is that if we create a new pointer and make a new malloc, we need to lose the `const` attribute because we need to free the memory.
So we create a const pointer and a non-const pointer. The const pointer takes the value whether of `index_sparse` or the new allocated pointer, and we can happily pass this `const` pointer to the `write` routines that expect a const pointer. But to free the memory, we use the non-const pointer.

Reply:
Thanks, now i see. Have you tested this for I/O of indices with large enough number of MOs (so that 32bit precision is actually used)? The unit tests use dummy 1000 mo-num so the 32bit code path is not tested.

Reply:
Tested in the last commit